services/html-help-service.ts

```diff
@@ -0,0 +1,161 @@
+///<reference path="../../.d.ts"/>
+"use strict";
+
+import util = require("util");
+import Future = require("fibers/future");
+import path = require("path");
+import options = require("options");
+import marked = require("marked");
+var TerminalRenderer = require('marked-terminal');
+var chalk = require("chalk");
+
+export class HtmlHelpService implements IHtmlHelpService {
+	private static MARKDOWN_FILE_EXTENSION = ".md";
+	private static HTML_FILE_EXTENSION = ".html";
+	private static MAN_PAGE_NAME_PLACEHOLDER = "@MAN_PAGE_NAME@";
+	private static HTML_COMMAND_HELP_PLACEHOLDER = "@HTML_COMMAND_HELP@";
+	private static RELATIVE_PATH_TO_STYLES_CSS_PLACEHOLDER = "@RELATIVE_PATH_TO_STYLES_CSS@";
+
+	private pathToManPages: string;
+	private pathToHtmlPages: string;
+	private get pathToStylesCss(): string {
+		return path.join(this.$staticConfig.HTML_HELPERS_DIR, "styles.css");
+	}
+
+	private get pathToBasicPage(): string {
+		return path.join(this.$staticConfig.HTML_HELPERS_DIR, "basic-page.html");
+	}
+
+	constructor(private $logger: ILogger,
+		private $injector: IInjector,
+		private $errors: IErrors,
+		private $fs: IFileSystem,
+		private $staticConfig: Config.IStaticConfig,
+		private $microTemplateService: IMicroTemplateService,
+		private $opener: IOpener) {
+		this.pathToHtmlPages = this.$staticConfig.HTML_PAGES_DIR;
+		this.pathToManPages = this.$staticConfig.MAN_PAGES_DIR;
+	}
+
+	public generateHtmlPages(): IFuture<void> {
+		return (() => {
+			var mdFiles = this.$fs.enumerateFilesInDirectorySync(this.pathToManPages);
+			var basicHtmlPage = this.$fs.readFile(this.pathToBasicPage).wait().toString();
+			var futures = _.map(mdFiles, markdownFile => this.createHtmlPage(basicHtmlPage, markdownFile));
+			Future.wait(futures);
+			this.$logger.trace("Finished generating HTML files.");
+		}).future<void>()();
+	}
+
+	private createHtmlPage(basicHtmlPage: string, pathToMdFile: string): IFuture<void> {
+		return (() => {
+			var mdFileName = path.basename(pathToMdFile);
+			var htmlFileName = mdFileName.replace(HtmlHelpService.MARKDOWN_FILE_EXTENSION, HtmlHelpService.HTML_FILE_EXTENSION);
+			this.$logger.trace("Generating '%s' help topic.", htmlFileName);
+
+			var helpText = this.$fs.readText(pathToMdFile).wait();
+			var outputText = this.$microTemplateService.parseContent(helpText, { isHtml: true });
+			var htmlText = marked(outputText);
+
+			var filePath = pathToMdFile
```

Comment:
Ahh, I've got it - as I want to preserve same directory structure, I need the replace method. If I use your suggestion, all html files will be placed inside html directory.